storage/queue.go

```diff
@@ -170,9 +179,26 @@ func (bq *baseQueue) Add(repl *Replica, priority float64) error {
 // not be added, as the replica with the lowest priority will be
 // dropped.
 func (bq *baseQueue) MaybeAdd(repl *Replica, now proto.Timestamp) {
+	// Load the system config.
+	cfg, err := bq.gossip.GetSystemConfig()
```

Comment:
The decoded system config should be cached internally to gossip; no need to expose the callback to callers like this.